kernel/comps/virtio/src/device/input/device.rs

```diff
@@ -167,25 +163,35 @@ impl InputDevice {
         }
     }
 
-    /// Query a specific piece of information by `select` and `subsel`, and write
-    /// result to `out`, return the result size.
-    pub fn query_config_select(&self, select: InputConfigSelect, subsel: u8, out: &mut [u8]) -> u8 {
+    pub fn query_config_id_name(&self) -> String {
+        let size = self.select_config(InputConfigSelect::IdName, 0);
+        let mut data_ptr = field_ptr!(&self.config, VirtioInputConfig, data);
+        let mut out = Vec::with_capacity(size);
+        for _ in 0..size {
+            out.push(data_ptr.clone().cast::<u8>().read_once().unwrap());
+            data_ptr.byte_add(1);
+        }
+        String::from_utf8(out).unwrap()
+    }
+
+    pub fn query_config_prop_bits(&self) -> InputProp {
+        let size = self.select_config(InputConfigSelect::PropBits, 0);
+        assert!(size <= 1);
+        let data_ptr = field_ptr!(&self.config, VirtioInputConfig, data);
+        InputProp::from_bits(data_ptr.clone().cast::<u8>().read_once().unwrap()).unwrap()
```

Comment:
We should not call `read_once' when the size is zero, right? I think we should return the empty prop directly.